### PR TITLE
feat(metrics): Add steps to the add_metrics method

### PR DESCRIPTION
**Pull request by lamroger**
Overview:
This allows users to pass a step value when using `add_metrics`. The step will be attached as a label and surfaced through the API and Frontend.

Testing:
Started a notebook on prod, used the terminal to pull down this branch to build and install.
Pushed metrics with steps and it was registered by the pushgateway.
Screen Shot 2020-12-04 at 2 49 06 PM
Screen Shot 2020-12-04 at 2 50 05 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -237,5 +238,8 @@ docker-compose build utils
 
 # To run tests
 docker-compose -f docker-compose.ci.yml run utils poetry run pytest
+
+# To autoformat
+docker-compose run utils poetry run autopep8 --in-place --aggressive --aggressive --recursive .
```
Comment (weikaolun) [inline README.md:243]: is this supposed to be extra-aggressive?

File: gradient_utils/metrics.py [hunk 2]
```diff
@@ -56,23 +58,27 @@ def get_workload_id():
...
+        step=None,
         timeout=30):
-    metrics_logger = MetricsLogger()
+    metrics_logger = MetricsLogger(step=step)
 
     metrics = [Metric(key, value) for key, value in metrics.items()]
     for metric in metrics:
         metrics_logger.add_gauge(metric.key)
+        logger.warning(metrics_logger[metric.key])
+        logger.warning(metric.value)
```
Comment (weikaolun) [inline gradient_utils/metrics.py:71]: drop or switch to debug?

File: README.md (hunk 1 above)
Comment (lamroger) [reply README.md:243]: yea - it was in the autopep8 example code. Figured it'd be harmless and extra formatted

File: gradient_utils/metrics.py (hunk 2 above)
Comment (lamroger) [reply gradient_utils/metrics.py]: switched to a debug statement

**Review by weikaolun — APPROVED** [review]

**Pull request merged**